### PR TITLE
Improve suggestion to change struct field to &mut

### PR DESCRIPTION
**Pull request by rukai**
r? @estebank 

Now displays a proper underline style suggestion instead of including the code change inline with the message.

### CODE REVIEW

**Comment by rukai** [comment]
r=estebank

File: compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs [hunk 1]
```diff
@@ -1081,25 +1081,12 @@ fn annotate_struct_field<'tcx>(
...
-                    } else {
-                        String::new()
-                    };
-
-                    return Some((
-                        field.ty.span,
-                        format!("&{}mut {}", lifetime_snippet, &*type_snippet,),
-                    ));
+                    return Some(
+                        lifetime.span.with_hi(lifetime.span.hi() + BytePos(1)).shrink_to_hi(),
```
Comment (estebank) [inline compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs:1088]:
```suggestion
                        lifetime.span.between(ty.span),
```

File: compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs [hunk 2]
```diff
@@ -1081,25 +1081,12 @@ fn annotate_struct_field<'tcx>(
             // Now we're dealing with the actual struct that we're going to suggest a change to,
             // we can expect a field that is an immutable reference to a type.
             if let hir::Node::Field(field) = node {
-                if let hir::TyKind::Rptr(
-                    lifetime,
-                    hir::MutTy { mutbl: hir::Mutability::Not, ref ty },
-                ) = field.ty.kind
+                if let hir::TyKind::Rptr(lifetime, hir::MutTy { mutbl: hir::Mutability::Not, .. }) =
```
Comment (estebank) [inline compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs:1084]:
```suggestion
                if let hir::TyKind::Rptr(lifetime, hir::MutTy { mutbl: hir::Mutability::Not, ty }) =
```

**Review by estebank — APPROVED** [review]
One nitpick

**Comment by rukai** [comment]
Ended up having to change some other things to get it working like that. Does this look good?

Also I'd like to know what the `r=me after rebasing` command does?

**Comment by estebank** [comment]
@bors r+ rollup

> Also I'd like to know what the r=me after rebasing command does?

you can do `@bors r=estebank` if you have permissions, but we also use `r=me` as a shorthand for "rebase and I'll approve when I see that you've done so, after you've rebased you can stop thinking about this PR".

**Pull request merged**